src/metricscollector/server/memory_metric_handler_test.go

```diff
@@ -106,4 +115,194 @@ var _ = Describe("MemoryMetricHandler", func() {
 			})
 		})
 	})
+
+	Describe("GetMemoryMetricHistory", func() {
+		JustBeforeEach(func() {
+			handler.GetMemoryMetricHistory(resp, req, map[string]string{"appid": "an-app-id"})
+		})
+
+		Context("when request query string is invalid", func() {
+			Context("when there are multiple start pararmeters in query string", func() {
+				BeforeEach(func() {
+					req, err = http.NewRequest(http.MethodGet, "http://localhost/v1/apps/an-app-id/metrics_history/memory?start=123&start=231", nil)
+					Expect(err).ToNot(HaveOccurred())
+
+				})
+
+				It("returns 400", func() {
+					Expect(resp.Code).To(Equal(http.StatusBadRequest))
+
+					errJson := &ErrorResponse{}
+					d := json.NewDecoder(resp.Body)
+					err = d.Decode(errJson)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(errJson.Code).To(Equal("Bad-Request"))
+					Expect(errJson.Message).To(Equal("Incorrect start parameter in query string"))
+				})
+			})
+
+			Context("when start time is not a number", func() {
+				BeforeEach(func() {
+					req, err = http.NewRequest(http.MethodGet, "http://localhost/v1/apps/an-app-id/metrics_history/memory?start=abc", nil)
+					Expect(err).ToNot(HaveOccurred())
+
+				})
+
+				It("returns 400", func() {
+					Expect(resp.Code).To(Equal(http.StatusBadRequest))
+
+					errJson := &ErrorResponse{}
+					d := json.NewDecoder(resp.Body)
+					err = d.Decode(errJson)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(errJson.Code).To(Equal("Bad-Request"))
+					Expect(errJson.Message).To(Equal("Error parsing start time"))
+				})
+			})
+
+			Context("when there are multiple end parameters in query string", func() {
+				BeforeEach(func() {
+					req, err = http.NewRequest(http.MethodGet, "http://localhost/v1/apps/an-app-id/metrics_history/memory?end=123&end=231", nil)
+					Expect(err).ToNot(HaveOccurred())
+
+				})
+
+				It("returns 400", func() {
+					Expect(resp.Code).To(Equal(http.StatusBadRequest))
+
+					errJson := &ErrorResponse{}
+					d := json.NewDecoder(resp.Body)
+					err = d.Decode(errJson)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(errJson.Code).To(Equal("Bad-Request"))
+					Expect(errJson.Message).To(Equal("Incorrect end parameter in query string"))
+				})
+			})
+
+			Context("when end time is not a number", func() {
+				BeforeEach(func() {
+					req, err = http.NewRequest(http.MethodGet, "http://localhost/v1/apps/an-app-id/metrics_history/memory?end=abc", nil)
+					Expect(err).ToNot(HaveOccurred())
+
+				})
+
+				It("returns 400", func() {
+					Expect(resp.Code).To(Equal(http.StatusBadRequest))
+
+					errJson := &ErrorResponse{}
+					d := json.NewDecoder(resp.Body)
+					err = d.Decode(errJson)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(errJson.Code).To(Equal("Bad-Request"))
+					Expect(errJson.Message).To(Equal("Error parsing end time"))
+				})
+			})
+
+		})
+
+		Context("when request query string is valid", func() {
```

Comment:
If you reordered your tests and got the base cases out of the way, you need to check less later on.
While it is good practice to make sure everything is valid, you can also get away with just checking for the difference/new behavior.